eras/conway/impl/src/Cardano/Ledger/Conway/TxBody.hs

```diff
@@ -1,145 +1,532 @@
+        ofield
+          (\x tx -> tx {ctbrVldt = (ctbrVldt tx) {invalidBefore = x}})
+          From
+      bodyFields 9 = field (\x tx -> tx {ctbrMint = x}) (D fromCBOR)
+      bodyFields 11 = ofield (\x tx -> tx {ctbrScriptIntegrityHash = x}) From
+      bodyFields 14 = field (\x tx -> tx {ctbrReqSignerHashes = x}) (D (decodeSet fromCBOR))
+      bodyFields 15 = ofield (\x tx -> tx {ctbrTxNetworkId = x}) From
+      bodyFields 19 = field (\x tx -> tx {ctbrGovActions = x}) (D (decodeStrictSeq fromCBOR))
+      bodyFields 20 = field (\x tx -> tx {ctbrVotes = x}) (D (decodeStrictSeq fromCBOR))
+      bodyFields n = field (\_ t -> t) (Invalid n)
+      requiredFields =
+        [ (0, "inputs"),
+          (1, "outputs"),
+          (2, "fee")
+        ]
+
+newtype ConwayTxBody era = TxBodyConstr (MemoBytes TxBodyRaw era)
+  deriving (Generic, SafeToHash, ToCBOR)
+
+instance (Era era, NoThunks (PParamsUpdate era)) => NoThunks (ConwayTxBody era)
+
+deriving instance
+  ( Era era,
+    Eq (Script era),
+    Eq (CompactForm (Value era)),
+    Eq (PParamsUpdate era)
+  ) =>
+  Eq (ConwayTxBody era)
+
+deriving newtype instance (Era era, NFData (PParamsUpdate era)) => NFData (ConwayTxBody era)
+
+deriving instance
+  ( Era era,
+    Val (Value era),
+    Show (Value era),
+    Show (Script era),
+    Show (PParamsUpdate era)
+  ) =>
+  Show (ConwayTxBody era)
+
+type instance MemoHashIndex TxBodyRaw = EraIndependentTxBody
+
+instance (c ~ EraCrypto era) => HashAnnotated (ConwayTxBody era) EraIndependentTxBody c where
+  hashAnnotated (TxBodyConstr mb) = getMemoBytesHash mb
+
+instance
+  ( Era era,
+    Val (Value era),
+    DecodeNonNegative (Value era),
+    FromCBOR (PParamsUpdate era),
+    FromCBOR (Annotator (Script era))
+  ) =>
+  FromCBOR (Annotator (TxBodyRaw era))
+  where
+  fromCBOR = pure <$> fromCBOR
+
+deriving via
+  (Mem TxBodyRaw era)
+  instance
+    ( Era era,
+      Val (Value era),
+      DecodeNonNegative (Value era),
+      FromCBOR (PParamsUpdate era),
+      FromCBOR (Annotator (Script era))
+    ) =>
+    FromCBOR (Annotator (ConwayTxBody era))
+
+mkConwayTxBodyFromRaw :: ConwayEraTxBody era => TxBodyRaw era -> ConwayTxBody era
+mkConwayTxBodyFromRaw = TxBodyConstr . memoBytes . encodeTxBodyRaw
+
+lensTxBodyRaw ::
+  ConwayEraTxBody era =>
+  (TxBodyRaw era -> a) ->
+  (TxBodyRaw era -> t -> TxBodyRaw era) ->
+  Lens (ConwayTxBody era) (ConwayTxBody era) a t
+lensTxBodyRaw getter setter =
+  lens
+    (\(TxBodyConstr (Memo txBodyRaw _)) -> getter txBodyRaw)
+    (\(TxBodyConstr (Memo txBodyRaw _)) val -> mkConwayTxBodyFromRaw $ setter txBodyRaw val)
+{-# INLINEABLE lensTxBodyRaw #-}
 
 instance CC.Crypto c => EraTxBody (ConwayEra c) where
   {-# SPECIALIZE instance EraTxBody (ConwayEra CC.StandardCrypto) #-}
 
-  type TxBody (ConwayEra c) = BabbageTxBody (ConwayEra c)
+  type TxBody (ConwayEra c) = ConwayTxBody (ConwayEra c)
 
-  mkBasicTxBody = mkBabbageTxBody
+  mkBasicTxBody = mkConwayTxBodyFromRaw initialTxBodyRaw
```

Comment:
Please refer to this PR to adjust they way that Memoized types should be defined: https://github.com/input-output-hk/cardano-ledger/pull/3165

In particular if you follow the diff that was applied to `Crdano.Ledger.Babbage.TxBody` module you'll see everything that needs to be changed